autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -272,6 +284,19 @@ def transform(
                                   "Please try to manually cast it to a supported "
                                   "numerical or categorical values.")
             raise e
+
+        if (
+            (
+                isinstance(X, np.ndarray) or scipy.sparse.issparse(X) or hasattr(X, 'iloc')
+            )
+            and self._dataset_compression is not None
+        ):
+            if self._precision is not None:
+                X = X.astype(self._precision)
+            else:
+                X = reduce_dataset_size_if_too_large(X, **self._dataset_compression)
+                self._precision = dict(X.dtypes) if hasattr(X, 'iloc') else X.dtype
+
         return X
```

Comment:
We should separate it as a method (btw, Does this method really return `np.ndarray`?)
```suggestion
    def _compress_dataset(self, X: np.ndarray) -> np.ndarray:
        is_dataframe = hasattr(X, 'iloc')
        is_reducible_type = isinstance(X, np.ndarray) or scipy.sparse.issparse(X) or is_dataframe
        if not is_reducible_type or self._dataset_compression is None:
            return X
        elif self._precision is not None:
            X = X.astype(self._precision)
            return X
        else:
            X = reduce_dataset_size_if_too_large(X, **self._dataset_compression)
            self._precision = dict(X.dtypes) if is_dataframe else X.dtype
            return X
```

Reply:
> We should separate it as a method (btw, Does this method really return `np.ndarray`?)

This points to a very important fact which I don't know how I missed. We would and should never get a pandas dataframe here. It will always be a numpy array or a sparse matrix.

Reply:
I have removed the code to compress a pandas dataframe. For our record, it will be there in the commit history.